libs/librepcbcommon/scopeguard.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2016 The LibrePCB developers
+ * http://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBREPCB_SCOPEGUARD_H
+#define LIBREPCB_SCOPEGUARD_H
+
+#include <utility>
+
+namespace librepcb {
+
+template<class Fun>
+class ScopeGuard
```

Comment:
If no method is declared virtual (not even the destructor), please use the "final" keyword to the class definition